espei/utils.py

```diff
@@ -46,19 +46,47 @@ def __setstate__(self, state):
         self.__init__(storage=MemoryStorage)
         self.insert_multiple(state['_tables']['_default'])
 
+class PredictWrapper:
+    def __init__(self, client, f, **kwargs):
+        self.f = f
+        self.client = client
+
+        # TODO: I'm debating whether to submit a self generating lambda function for the context or to scatter the context
+        # as data. With scattering, the data will transfer before the worker starts anything, but I don't know
+        # if a worker that restarts will retain the data. The lambda function will transfer data after the worker starts
+        # (it should only be once at the start), but I think if a worker dies, it could still generate the data again (although,
+        # since the self generating function lives on a specific worker, I don't know what would happen if that specific worker
+        # restarts. Memory and performance-wise, scattering vs submitting lambda function seems to be the same
+        self.kwargs = {key: self.client.submit(lambda x: x, val, key=key) for key,val in kwargs.items()}
+        #self.kwargs = {key: self.client.scatter(val, broadcast=True, hash=False) for key,val in kwargs.items()}
+    
+    def __call__(self, x):
+        return self.client.submit(self.f, x, **self.kwargs)
 
 class ImmediateClient(Client):
     """
     A subclass of distributed.Client that automatically unwraps the Futures
     returned by map.
     """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # Not really a fan of disabling the active memory manager (AMM), although it's somewhat experimental in dask
+        # However, the AMM will delete any duplicates found across workers. The consequence of this is when
+        # the context is created on all workers, the AMM will remove all duplicates of the context except for 
+        # one worker, and all the dependent tasks will then be done on that single worker (and we won't see any benefit
+        # from parallelization).
```

Comment:
The way this reads, we don't have a choice but to disable AMM, right?

Reply:
I changed the comment to just be an explanation of why we disable it.